dts/bindings/base/pm.yaml

```diff
@@ -30,3 +30,8 @@ properties:
     description: |
       Automatically configure the device for runtime power management after the
       init function runs.
+
+  zephyr,disabling-power-states:
+    type: phandles
+    description: |
+      List of power states that will disable this device power.
```

Comment:
Should this encompass states in which devices lose power **by definition** ?

I already had a discussion on this point [here](https://github.com/zephyrproject-rtos/zephyr/pull/60369#discussion_r1305436477). By definition of s2ram, most devices will lose power in this state.

To me `zephyr,disabling-power-states` should only document s2idle states.
Describing in a device node that it is loosing power in standby state doesn't make sense to me

Reply:
> By definition of s2ram, most devices will lose power in this state.

"most" is the reason for creating this. Assuming a soc has 3 s2ram substates defined, a device can lose power in 2 of them. Another device can lose power in one, while yet another instance of the same device in all 3. We need a way to clearly define what set of devices is covered by "most" and act accordingly in the device drivers.

One scenario we need to cover is nRF54H20 where almost all MVDMA instances lose power in s2ram substates 0 and 1, but there is one instance that loses power in s2ram substate 1 only. Without this PR we could not use the special MVDMA instance while a CPU is in s2ram. This would unnecessarily limit the available system features.

Reply:
Then what about combining information from each state and information in specific impacted nodes ?

Note that if this is merged, on STM32, we'll need to add this property in > 1000 device tree nodes, across 10th of device drivers types. With the involved possiblilities of errors.
This is not a minor thing.

Hence I'd like to be really sure that all other ideas have been carefully studied.

Reply:
Note also, that instead of listing all impacted states, we can mention the "highest" low power state causing this. I haven't seen cases where clocks/power could be lost in one state and preserved in the next lower low power state

Reply:
> Note that if this is merged, on STM32, we'll need to add this property in > 1000 device tree nodes, across 10th of device drivers types. With the involved possiblilities of errors. This is not a minor thing.

I understand the new property is optional. If you don't intend to use the framework introduced in this PR, you don't need to update the device tree nodes. When you decide to start using this framework, you can start updating the device drivers one by one, each modification associated with updating all compatible device tree nodes.

> Note also, that instead of listing all impacted states, we can mention the "highest" low power state causing this. I haven't seen cases where clocks/power could be lost in one state and preserved in the next lower low power state

I agree. I also haven't seen anything like this. Moreover, if we used only the "highest" power state in the device tree, it would make adding power states easier (fewer device tree nodes to update).

Reply:
> > Note that if this is merged, on STM32, we'll need to add this property in > 1000 device tree nodes, across 10th of device drivers types. With the involved possiblilities of errors. This is not a minor thing.
> 
> I understand the new property is optional. If you don't intend to use the framework introduced in this PR, you don't need to update the device tree nodes. When you decide to start using this framework, you can start updating the device drivers one by one, each modification associated with updating all compatible device tree nodes.
> 
> > Note also, that instead of listing all impacted states, we can mention the "highest" low power state causing this. I haven't seen cases where clocks/power could be lost in one state and preserved in the next lower low power state
> 
> I agree. I also haven't seen anything like this. Moreover, if we used only the "highest" power state in the device tree, it would make adding power states easier (fewer device tree nodes to update).

What "highest" means here ? The deepest state that the SoC can go without loosing power ?

Reply:
> > > Note also, that instead of listing all impacted states, we can mention the "highest" low power state causing this. I haven't seen cases where clocks/power could be lost in one state and preserved in the next lower low power state
> > 
> > 
> > I agree. I also haven't seen anything like this. Moreover, if we used only the "highest" power state in the device tree, it would make adding power states easier (fewer device tree nodes to update).
> 
> What "highest" means here ? The deepest state that the SoC can go without loosing power ?

The most natural way of expressing this seems to be the lightest sleep that disables power (or clock) of a device. And a way for a device driver to lock this and all deeper sleep sub-states and states.